pkg/ccl/crosscluster/logical/logical_replication_writer_processor.go

```diff
@@ -491,7 +501,16 @@ func (lrw *logicalReplicationWriterProcessor) flushBuffer(
 				if err != nil {
 					// TODO(ssd): Handle errors. We should perhaps split the batch and retry a portion of the batch.
 					// If that fails, send the failed application to the dead-letter-queue.
-					return err
+
+					// TODO(azhu): Remove these nil variables once refactoring is done
+					// & we've implemented conflict handling logic.
+					var kv roachpb.KeyValue
+					var cdcEventRow cdcevent.Row
+					var writeTimestamp hlc.Timestamp
+					var conflictReason string
+					if err = lrw.dlqClient.Log(ctx, lrw.spec.JobID, kv, cdcEventRow, writeTimestamp, conflictReason); err != nil {
+						return err
```

Comment:
note that you are swallowing the parent error here! I'd return `errors.Combine()` to make sure the client sees the og error and the DLQ log error.